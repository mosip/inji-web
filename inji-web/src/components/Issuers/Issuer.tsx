import React from "react";
import {getIssuerDisplayObjectForCurrentLanguage} from "../../utils/i18n";
import {useNavigate} from "react-router-dom";
import {ItemBox} from "../Common/ItemBox";
import {IssuerProps} from "../../types/components";
import {useSelector} from "react-redux";
import {RootState} from "../../types/redux";
import { IssuerWellknownDisplayArrayObject } from "../../types/data";

export const Issuer: React.FC<IssuerProps> = ({issuer, index}) => {
    const language = useSelector((state: RootState) => state.common.language);
    const issuerDisplayObject: IssuerWellknownDisplayArrayObject = getIssuerDisplayObjectForCurrentLanguage(issuer.display, language);
    const navigate = useNavigate();
    return <React.Fragment>
        <ItemBox index={index}
                 url={issuerDisplayObject?.logo?.url}
                 title={issuerDisplayObject?.name}
                 description={issuerDisplayObject?.description}
                 onClick={() => navigate(`/issuers/${issuer.issuer_id}`)}/>
    </React.Fragment>
}

