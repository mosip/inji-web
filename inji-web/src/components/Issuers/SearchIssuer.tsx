import React, {useState} from "react";
import {useTranslation} from "react-i18next";
import {FaSearch} from 'react-icons/fa';
import {useDispatch, useSelector} from "react-redux";
import {storeFilteredIssuers} from "../../redux/reducers/issuersReducer";
import {IssuerObject} from "../../types/data";
import {IoCloseCircleSharp} from "react-icons/io5";
import {SearchIssuerProps} from "../../types/components";
import {RootState} from "../../types/redux";
import {getObjectForCurrentLanguage} from "../../utils/i18n";

export const SearchIssuer: React.FC<SearchIssuerProps> = (props) => {

    const {t} = useTranslation("HomePage");
    const dispatch = useDispatch();
    const [searchText, setSearchText] = useState("");
    const issuers = useSelector((state:RootState) => state.issuers.issuers);
    const language = useSelector((state:RootState) => state.common.language);
    const filterIssuers = async (searchText: string) => {
        setSearchText(searchText);
        const filteredIssuers = issuers.filter( (issuer:IssuerObject) => {
            const displayObject = getObjectForCurrentLanguage(issuer.display, language);
            return (displayObject.name.toLowerCase().indexOf(searchText.toLowerCase()) !== -1 && issuer.protocol !== 'OTP')
        })
        dispatch(storeFilteredIssuers(filteredIssuers));
    }

    return <React.Fragment>
        <div className={"flex justify-center items-center w-full"}>
            <div data-testid="Search-Issuer-Container" className="sm:w-8/12 w-full flex justify-start items-center bg-iw-background shadow-md shadow-iw-shadow">
                <FaSearch data-testid="Search-Issuer-Search-Icon"
                          color={'var(--iw-color-searchIcon)'}
                          className={"m-5"}
                          size={20}/>
                <input
                    data-testid="Search-Issuer-Input"
                    type="text"
                    value={searchText}
                    placeholder={t("Intro.searchText")}
                    onChange={event => filterIssuers(event.target.value)}
                    className="py-6 w-11/12 text-iw-searchTitle focus:outline-none overflow-ellipsis mr-10"
                />
                {searchText.length > 0 && <IoCloseCircleSharp data-testid="Search-Issuer-Clear-Icon"
                                                              onClick={() => filterIssuers("")}
                                                              className={"m-5"}
                                                              color={'var(--iw-color-closeIcon)'}
                                                              size={20}/>}
            </div>
        </div>
    </React.Fragment>
}

