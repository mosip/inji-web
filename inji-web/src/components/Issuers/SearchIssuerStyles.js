export const SearchIssuerStyles = {
  infoMsg: "text-iw-closeIcon text-sm-plus",
  regexError: "text-iw-red text-sm-plus"
};