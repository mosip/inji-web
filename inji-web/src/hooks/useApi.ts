import {useState} from "react";
import {ApiRequest} from "../types/data";
import {api, MethodType} from "../utils/api";

export interface NetworkResult<T> {
    data: T | null;
    error: Error | null;
    status: number | null;
    loading: boolean;
    headers: object;
}

interface RequestConfig {
    url?: string;
    headers?: Record<string, string>;
    body?: any;
    apiRequest: ApiRequest;
}

interface UseApiReturn<T> {
    data: T | null;
    error: Error | null;
    loading: boolean;
    status: number | null;
    fetchData: (arg0: RequestConfig) => Promise<NetworkResult<T>>;
}

export function useApi<T = any>(): UseApiReturn<T> {
    const [data, setData] = useState<T | null>(null);
    const [error, setError] = useState<Error | null>(null);
    const [loading, setLoading] = useState<boolean>(false);
    const [status, setStatus] = useState<number | null>(null);

    async function fetchData({
                                 headers,
                                 body,
                                 apiRequest,
                                 url = undefined
                             }: RequestConfig): Promise<NetworkResult<T>> {
        setLoading(true);
        setError(null);
        setStatus(null);

        let result: NetworkResult<T> = {
            data: null,
            error: null,
            status: null,
            loading: true,
            headers: {}
        };

        try {
            const response = await api.instance.request({
                url: url ?? apiRequest.url(),
                method: MethodType[apiRequest.methodType],
                headers,
                data: body,
                withCredentials: apiRequest.credentials === "include",
                responseType: apiRequest.responseType ?? "json",
            });

            console.log("API response:", response);
            setData(response.data);
            setStatus(response.status);

            result = {
                data: response.data,
                error: null,
                status: response.status,
                loading: false,
                headers: response.headers || {}
            };
        } catch (err: any) {
            const parsedError = err instanceof Error ? err : new Error("Unknown error");

            setError(parsedError);
            setStatus(err?.response?.status || null);

            result = {
                data: null,
                error: parsedError,
                status: err?.response?.status || null,
                loading: false,
                headers: err?.response?.headers || {}
            };
        } finally {
            setLoading(false);
        }

        return result;
    }

    return {data, error, loading, status, fetchData};
}
