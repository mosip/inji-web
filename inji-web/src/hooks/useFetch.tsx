import {useState} from "react";
import {api, MethodType} from "../utils/api";

export enum RequestStatus {
    LOADING,
    DONE,
    ERROR
}

export const useFetch = () => {
    const [state, setState] = useState<RequestStatus>(RequestStatus.LOADING);
    const [error, setError] = useState<string>("");

    const fetchRequest = async (uri: string, method: MethodType, header: any, body?: any) => {
        try {
            setState(RequestStatus.LOADING);
            const response = await fetch(`${uri}`, {
                method: MethodType[method],
                headers: header,
                body: body,
            });

            if (uri.indexOf("download") !== -1) {
                setState(RequestStatus.DONE);
                return await response.blob();
            }
            if (!response.ok) {
                setState(RequestStatus.ERROR);
                return response;
            }
            setState(RequestStatus.DONE);
            return await response.json();
        } catch (e) {
            setState(RequestStatus.ERROR);
            setError("Error Happened");
        }
    };
    return {state, error, fetchRequest};
}



