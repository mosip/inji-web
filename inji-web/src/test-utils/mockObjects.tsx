import { IssuerConfigurationObject } from '../types/data';
import { LandingPageWrapperProps } from '../components/Common/LandingPageWrapper';
import { NavBarProps } from '../types/components';
export const mockIssuerDisplayArrayObject = {
    name: "Name",
    language: "en",
    locale: "en",
    logo: {
        url: "https://url.com",
        alt_text: "alt text of the url"
    },
    title: "Title",
    description: "Description",
};

export const mockCredentialTypeDisplayArrayObject = {
    name: "Name",
    locale: "en",
    logo: "https://url.com",
};

export const mockCredentials: IssuerConfigurationObject = {
    credentials_supported: [
        {
            name: "InsuranceCredential",
            scope: "mosip_vc_ldp",
            display: [
                {
                    name: "Health Insurance",
                    locale: "en",
                    logo: "https://url.com"
                }
            ]
        },
        {
            name: "AnotherCredential",
            scope: "mosip_ldp_vc",
            display: [
                {
                    name: "Another Credential",
                    locale: "en",
                    logo: "https://url.com"
                }
            ]
        }
    ],
    "authorization_endpoint": "https://env.net/authorize",
    "grant_types_supported": ["authorization_code"]
};


export const mockIssuerObject = {
        name: 'Test Issuer',
        desc: 'Test Description',
        protocol: 'OTP' as 'OTP', // Explicitly set the type to 'OTP' or 'OpenId4VCI'
        issuer_id: 'test-issuer',
        authorization_endpoint: 'https://auth.test.com',
        credentials_endpoint: 'https://credentials.test.com',
        display: [{
            name: "Name",
            language: "en",
            locale: "en",
            logo: {
                url: "https://url.com",
                alt_text: "alt text of the url"
            },
            title: "Title",
            description: "Description",
        }],
        client_id: 'test-client-id',
        redirect_uri:'test-redirect-uri',
        token_endpoint:'test-token_endpoint',
        proxy_token_endpoint:'test-proxy_token_endpoint',
        client_alias:'',
        ovp_qr_enabled: true,
        scopes_supported: ['scope1', 'scope2']
}

export const mockIssuerObjectList =  [
    {
        name: 'Issuer 1',
        desc: 'Description 1',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer1',
        authorization_endpoint: 'https://issuer1.com/auth',
        credentials_endpoint: 'https://issuer1.com/credentials',
        display: [
            {
                name: 'Issuer 1',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer1.com/logo.png', alt_text: 'Issuer 1 Logo' },
                title: 'Issuer 1 Title',
                description: 'Issuer 1 Description',
            },
        ],
        client_id: 'client1',
        redirect_uri: 'https://issuer1.com/redirect',
        token_endpoint: 'https://issuer1.com/token',
        proxy_token_endpoint: 'https://issuer1.com/proxy-token',
        client_alias: 'client1-alias',
        ovp_qr_enabled: true,
        scopes_supported: ['openid', 'profile'],
    },
    {
        name: 'Issuer 2',
        desc: 'Description 2',
        protocol: 'OpenId4VCI',
        issuer_id: 'issuer2',
        authorization_endpoint: 'https://issuer2.com/auth',
        credentials_endpoint: 'https://issuer2.com/credentials',
        display: [
            {
                name: 'Issuer 2',
                language: 'en',
                locale: 'en-US',
                logo: { url: 'https://issuer2.com/logo.png', alt_text: 'Issuer 2 Logo' },
                title: 'Issuer 2 Title',
                description: 'Issuer 2 Description',
            },
        ],
        client_id: 'client2',
        redirect_uri: 'https://issuer2.com/redirect',
        token_endpoint: 'https://issuer2.com/token',
        proxy_token_endpoint: 'https://issuer2.com/proxy-token',
        client_alias: 'client2-alias',
        ovp_qr_enabled: false,
        scopes_supported: ['openid', 'profile'],
    },
]

export const mockLandingPageWrapperProps: LandingPageWrapperProps = {
    icon: <div data-testid="Test-Icon">Icon</div>,
    title: "Test Title",
    subTitle: "Test SubTitle",
    gotoHome: true,
};

export const mockNavBarProps: NavBarProps = {
    title: 'Test Title',
    link: '/test-link',
    search: true,
};