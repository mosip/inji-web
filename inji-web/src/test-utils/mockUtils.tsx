import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, BrowserRouter as Router, Routes } from 'react-router-dom';
import { reduxStore } from '../redux/reduxStore';

// Mock for storage module
export const mockStorageModule = () => {
    jest.mock('../utils/storage.ts', () => ({
        storage: {
            getItem: jest.fn(),
            setItem: jest.fn(),
            SESSION_INFO: 'SESSION_INFO',
            SELECTED_LANGUAGE: 'selectedLanguage'
        }
    }));
};

// Mock for localStorage
export const mockLocalStorage = () => {
    let store: { [key: string]: string } = {};
  
    const localStorageMock = {
      getItem: jest.fn((key: string) => store[key] || null),
      setItem: jest.fn((key: string, value: string) => {
        store[key] = value;
      })
    };
  
    Object.defineProperty(window, 'localStorage', { 
      value: localStorageMock,
      writable: true
    });
  
    return localStorageMock;
  };

// API Mock
export const mockApi = {
    authorizationRedirectionUrl: 'http://mockurl.com'
};

// Crypto Mock
export const mockCrypto = {
    getRandomValues: (array: Uint32Array) => {
        for (let i = 0; i < array.length; i++) {
            array[i] = 4294967296;
        }
        return array;
    },
    subtle: {} as SubtleCrypto,
    randomUUID: () => '123e4567-e89b-12d3-a456-426614174000'
} as Crypto;

export const mockUseTranslation = () => {
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
            t: (key: string) => key,
        }),
    }));
};

export const mockUseNavigate = () => {
    const mockNavigate = jest.fn();
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useNavigate: mockNavigate,
    }));
};

export const mockUseSelector = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useSelector: jest.fn(),
    }));
};

export const setMockUseSelectorState = (state: any) => {
    const useSelectorMock = require('react-redux').useSelector;
    useSelectorMock.mockImplementation((selector: any) => selector(state));
};

export const mockUseGetObjectForCurrentLanguage = () => {
    jest.mock('../utils/i18n', () => ({
        getObjectForCurrentLanguage: jest.fn(),
    }));
};

export const wrapUnderRouter = (children: React.ReactNode) => {
    return <Router>{children}</Router>;
};

export const mockUseSearchCredentials = () => {
    jest.mock('../components/Credentials/SearchCredential', () => ({
        SearchCredential: () => <></>,
    }));
 };

export const mockUseDispatch = () => {
    jest.mock('react-redux', () => ({
        ...jest.requireActual('react-redux'),
        useDispatch: jest.fn(),
    }));
};
              
export const mockUseSpinningLoader = () => {
    jest.mock('../components/Common/SpinningLoader', () => ({
        SpinningLoader: () => <></>,
    }));
};

export const mockUseLanguageSelector = () => {
    jest.mock('../components/Common/LanguageSelector', () => ({
        LanguageSelector: () => <></>,
    }));
};

interface RenderWithProviderOptions extends Omit<RenderOptions, 'queries'> {}

export const renderWithProvider = (element: ReactElement, options?: RenderWithProviderOptions) => {
    return render(
        <Provider store={reduxStore}>
            <Router>
                {element}
            </Router>
        </Provider>,
        options
    );
};
    
export const mockUseParams = ()=>{
    jest.mock('react-router-dom', () => ({
        ...jest.requireActual('react-router-dom'),
        useParams: () => ({ issuerId: 'test-issuer-id' }),
      }));   
};

export const mockUseapiObject = () =>{
    jest.mock('../utils/api', () => ({
        api: {
          mimotoHost: 'http://mocked-api-host',
        },
      }));
};

export const mockUseFetch = () =>{
    jest.mock('../hooks/useFetch', () => {
        const RequestStatus = {
          LOADING: 'LOADING',
          DONE: 'DONE',
          ERROR: 'ERROR',
        };
        return {
          useFetch: () => jest.fn(),
          RequestStatus,
        };
      });
};

export const mockUseToast = () =>{
    jest.mock('react-toastify', () => ({
      toast: {
        error: jest.fn(),
      },
    }));
};

export const mockusemisc = ()=>{
    jest.mock('../utils/misc', () => ({
        downloadCredentialPDF: jest.fn(),
        getErrorObject: jest.fn(),
        getTokenRequestBody: jest.fn(),
      }));
};

export const mockusei18n = ()=>{
    jest.mock('react-i18next', () => ({
        useTranslation: () => ({
          t: (key: string) => key,
        }),
        initReactI18next: {
          type: '3rdParty',
          init: jest.fn(),
        },
      }));

};

export const mockWindowLocation = (url: string) => {
    const location = new URL(url) as unknown as Location;
    Object.defineProperty(window, 'location', {
      value: location,
      writable: true,
    });
  };

export const renderWithRouter = (Element: React.ReactElement, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(
      <BrowserRouter>
        <Provider store={reduxStore}>
          <Routes>
            <Route path="*" element={Element} />
          </Routes>
        </Provider>
      </BrowserRouter>
    );
  };

