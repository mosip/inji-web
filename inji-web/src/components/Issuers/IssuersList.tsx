import React from "react";
import {Issuer} from "./Issuer";
import {useSelector} from "react-redux";
import {IssuerObject} from "../../types/data";
import {RootState} from "../../types/redux";
import {useTranslation} from "react-i18next";
import {EmptyListContainer} from "../Common/EmptyListContainer";
import {RequestStatus} from "../../hooks/useFetch";
import {SpinningLoader} from "../Common/SpinningLoader";
import {IssuersListProps} from "../../types/components";
import {HeaderTile} from "../Common/HeaderTile";

export const IssuersList: React.FC<IssuersListProps> = ({state}) => {
    const issuers = useSelector((state: RootState) => state.issuers);
    const {t} = useTranslation("HomePage");

    if (state === RequestStatus.LOADING) {
        return <SpinningLoader/>
    }

    if(state === RequestStatus.ERROR || !issuers?.filtered_issuers || (issuers?.filtered_issuers && issuers?.filtered_issuers?.length === 0)) {
        return <div data-testid="Issuers-List-Container">
            <HeaderTile content={t("containerHeading")}/>
            <EmptyListContainer content={t("emptyContainerContent")}/>
        </div>
    }

    return <React.Fragment>
        <div data-testid="Issuers-List-Container" className={"mb-20"}>
            <HeaderTile content={t("containerHeading")}/>
            <div className="grid grid-cols-1 md:grid-cols-3 sm:grid-cols-2 gap-5">
                {issuers.filtered_issuers.map((issuer: IssuerObject, index: number) =>
                    <Issuer issuer={issuer} key={index} index={index}/>)}
            </div>
        </div>
    </React.Fragment>
}

