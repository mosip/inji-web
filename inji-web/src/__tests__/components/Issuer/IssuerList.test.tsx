import { IssuersList } from '../../../components/Issuers/IssuersList';
import { RequestStatus } from '../../../hooks/useFetch';
import { mockIssuerObjectList } from '../../../test-utils/mockObjects';
import { renderWithProvider,mockUseSelector,mockUseTranslation, mockCrypto } from '../../../test-utils/mockUtils';

mockUseTranslation();
mockUseSelector();
global.crypto = mockCrypto;

describe("Testing the Layout of IssuersList", () => {
    beforeEach(() => {
        const useSelectorMock = require('react-redux').useSelector;
        useSelectorMock.mockImplementation((selector: any) => selector({
            issuers: {
                issuers: mockIssuerObjectList,
                filtered_issuers: mockIssuerObjectList,
            },
            common: {
                language: 'en',
            },
        }));
    });

    test('Check whether it renders loading state', () => {
        
        const { asFragment } = renderWithProvider(<IssuersList state={RequestStatus.LOADING} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check whether it renders error state', () => {
        const { asFragment } = renderWithProvider(<IssuersList state={RequestStatus.ERROR}/>);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check whether it renders empty issuers list', () => {
        const useSelectorMock = require('react-redux').useSelector;
        useSelectorMock.mockImplementation((selector: any) => selector({
            issuers: {
                issuers: [],
                filtered_issuers: [],
            },
            common: {
                language: 'en',
            },
        }));

        const { asFragment } = renderWithProvider(<IssuersList state={RequestStatus.DONE} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Check whether it renders issuers list properly', () => {
        const { asFragment } = renderWithProvider(<IssuersList state={RequestStatus.DONE} />);
        // expect(screen.getByText('Issuer 1')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });
    afterEach(() => {
        jest.clearAllMocks();
    });
});
