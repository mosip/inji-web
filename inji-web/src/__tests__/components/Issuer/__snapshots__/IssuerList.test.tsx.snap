// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layout of IssuersList Check whether it renders empty issuers list 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col items-center justify-center"
    data-testid="Issuers-List-Container"
  >
    <div
      class="font-bold text-iw-title mt-8 text-xl flex"
      data-testid="HeaderTile-Text"
    >
      List of Issuers
    </div>
    <div
      class="text-iw-subTitle text-md mb-8 mt-2 flex"
      data-testid="HeaderTile-Text-SubContent"
    >
      Search for your trusted issuer and choose a credential type in the next step.
    </div>
    <div
      class="flex justify-center items-center w-full mx-auto bg-iw-background my-auto flex-col h-72 shadow-md shadow-iw-shadow border-t-iw-shadow px-10"
      data-testid="EmptyList-Outer-Container"
    >
      <p
        class="text-center text-light-title dark:text-dark-title"
        data-testid="EmptyList-Text"
      >
        No issuers found. Please refresh your browser window or try again later
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of IssuersList Check whether it renders error state 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col items-center justify-center"
    data-testid="Issuers-List-Container"
  >
    <div
      class="font-bold text-iw-title mt-8 text-xl flex"
      data-testid="HeaderTile-Text"
    >
      List of Issuers
    </div>
    <div
      class="text-iw-subTitle text-md mb-8 mt-2 flex"
      data-testid="HeaderTile-Text-SubContent"
    >
      Search for your trusted issuer and choose a credential type in the next step.
    </div>
    <div
      class="flex justify-center items-center w-full mx-auto bg-iw-background my-auto flex-col h-72 shadow-md shadow-iw-shadow border-t-iw-shadow px-10"
      data-testid="EmptyList-Outer-Container"
    >
      <p
        class="text-center text-light-title dark:text-dark-title"
        data-testid="EmptyList-Text"
      >
        No issuers found. Please refresh your browser window or try again later
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of IssuersList Check whether it renders issuers list properly 1`] = `
<DocumentFragment>
  <div
    class=" flex flex-col items-center justify-center"
    data-testid="Issuers-List-Container"
  >
    <div
      class="font-bold text-iw-title mt-8 text-xl flex"
      data-testid="HeaderTile-Text"
    >
      List of Issuers
    </div>
    <div
      class="text-iw-subTitle text-md mb-8 mt-2 flex"
      data-testid="HeaderTile-Text-SubContent"
    >
      Search for your trusted issuer and choose a credential type in the next step.
    </div>
    <div
      class="flex justify-center items-center w-full mx-auto bg-iw-background my-auto flex-col h-72 shadow-md shadow-iw-shadow border-t-iw-shadow px-10"
      data-testid="EmptyList-Outer-Container"
    >
      <p
        class="text-center text-light-title dark:text-dark-title"
        data-testid="EmptyList-Text"
      >
        No issuers found. Please refresh your browser window or try again later
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the Layout of IssuersList Check whether it renders loading state 1`] = `
<DocumentFragment>
  <svg
    height="0"
    width="0"
  >
    <lineargradient
      id="blue-gradient-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
      x1="0%"
      x2="100%"
      y1="0%"
      y2="100%"
    >
      <stop
        offset="0%"
        stop-color="var(--iw-color-primary)"
      />
      <stop
        offset="100%"
        stop-color="var(--iw-color-secondary)"
      />
    </lineargradient>
  </svg>
  <div
    class="flex justify-center items-center"
    data-testid="SpinningLoader-Container"
  >
    <div
      aria-busy="true"
      aria-label="oval-loading"
      class="sc-beySPh gZmPOF"
      data-testid="oval-loading"
      role="progressbar"
    >
      <svg
        data-testid="oval-svg"
        height="80"
        stroke="var(--iw-color-spinningLoaderPrimary)"
        viewBox="-20.5 -20.5 43 43"
        width="80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <g
            data-testid="oval-secondary-group"
            stroke-width="3"
            transform="translate(1 1)"
          >
            <circle
              cx="0"
              cy="0"
              r="20"
              stroke="var(--iw-color-spinningLoaderSecondary)"
              stroke-opacity=".5"
              stroke-width="3"
            />
            <path
              d="M20 0c0-9.94-8.06-20-20-20"
            >
              <animatetransform
                attributeName="transform"
                dur="1s"
                from="0 0 0"
                repeatCount="indefinite"
                to="360 0 0"
                type="rotate"
              />
            </path>
          </g>
        </g>
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
